### PR TITLE
[All] Update to collectAsStateWithLifecycle to latest alpha

### PR DESCRIPTION
**Pull request by astamato**

### CODE REVIEW

**Comment by astamato** [comment]
This version of the lifecycle:lifecycle library brings a beta[!!!] compose UI and compiler version.

+--- androidx.lifecycle:lifecycle-runtime-compose:2.6.0-alpha02
|    +--- androidx.annotation:annotation-experimental:1.1.0
|    +--- androidx.compose.runtime:runtime:1.0.1 -> 1.3.0-beta03 (*)
|    +--- androidx.compose.ui:ui:1.0.1 -> 1.3.0-beta03 (*)
|    \--- androidx.lifecycle:lifecycle-runtime-ktx:2.6.0-alpha02 (*)

File: Jetcaster/README.md [hunk 1]
```diff
@@ -64,11 +64,11 @@ Using the example of the home screen in the [`com.example.jetcaster.ui.home`](ap
 
  - The ViewModel is implemented as [`HomeViewModel`][homevm], which exposes a `StateFlow<HomeViewState>` for the UI to observe.
  - [`HomeViewState`][homevm] contains the complete view state for the home screen as an [`@Immutable`](https://developer.android.com/reference/kotlin/androidx/compose/runtime/Immutable) `data class`.
- - The Home Compose UI in [`Home.kt`][homeui] uses [`HomeViewModel`][homevm], and observes it's [`HomeViewState`][homevm] as Compose [State](https://developer.android.com/reference/kotlin/androidx/compose/runtime/State), using [`collectAsState()`](https://developer.android.com/reference/kotlin/androidx/compose/package-summary#collectasstate):
+ - The Home Compose UI in [`Home.kt`][homeui] uses [`HomeViewModel`][homevm], and observes it's [`HomeViewState`][homevm] as Compose [State](https://developer.android.com/reference/kotlin/androidx/compose/runtime/State), using [`collectAsStateWithLifecycle()`](https://developer.android.com/reference/kotlin/androidx/compose/package-summary#collectasstate):
```
Comment (manuelvicnt) [inline Jetcaster/README.md:67]:
```suggestion
 - The Home Compose UI in [`Home.kt`][homeui] uses [`HomeViewModel`][homevm], and observes it's [`HomeViewState`][homevm] as Compose [State](https://developer.android.com/reference/kotlin/androidx/compose/runtime/State), using [`collectAsStateWithLifecycle()`](https://developer.android.com/reference/kotlin/androidx/lifecycle/compose/package-summary#(kotlinx.coroutines.flow.StateFlow).collectAsStateWithLifecycle(androidx.lifecycle.LifecycleOwner,androidx.lifecycle.Lifecycle.State,kotlin.coroutines.CoroutineContext)):
```

**Review by manuelvicnt — APPROVED** [review]

File: Jetchat/app/src/main/java/com/example/compose/jetchat/data/FakeData.kt [hunk 2]
```diff
@@ -40,7 +40,7 @@ private val initialMessages = listOf(
     ),
     Message(
         "Taylor Brooks",
-        "@aliconors Take a look at the `Flow.collectAsState()` APIs",
+        "@aliconors Take a look at the `Flow.collectAsStateWithLifecycle()` APIs",
```
Comment (bentrengrove) [inline Jetchat/app/src/main/java/com/example/compose/jetchat/data/FakeData.kt:43]: I like that you updated this

**Review by bentrengrove — APPROVED** [review]
LGTM

**Pull request merged**